record_adb.py

```diff
@@ -0,0 +1,48 @@
+import sys
+import argparse
+import subprocess
+import time
+import os
+import signal
+
+def main(args):
+    recording_name = ''
+    method = args.input
+    recording_name = './sdcard/test.mp4'
+    if args.output is not None:
+            recording_name = './sdcard/' + args.output
+    if method is 'touch' and args.coordinate_x is None or args.coordinate_y is None:
+        print('USE: -i touch --coordinate-x <coordinate> --coordinate-y <coordinate> to use the touch input')
+        sys.exit()
+    elif method is 'intent':
+        record_with_intent(recording_name)
+    else:
+        simulate_input(args.coordinate_x, args.coordinate_y, recording_name)
+        exit()
+
+
+def record_with_intent(recording_name):
+    record_process = subprocess.Popen(['adb', 'shell','screenrecord'] + [recording_name])
+    time.sleep(3)
+    activity_start = subprocess.Popen(['adb', 'shell', 'am', 'start-activity', 'org.mozilla.fenix.nightly/org.mozilla.fenix.HomeActivity', '--ez finishonboarding true'])
+    pull_recording(recording_name)
+
+def simulate_input(x,y, recording_name):
+    record_process = subprocess.Popen(['adb', 'shell','screenrecord'] + [recording_name])
+    time.sleep(5)
+    proc = subprocess.Popen(['adb', 'shell', 'input', 'tap'] + [str(x),str(y)])
+    pull_recording(recording_name)
+
+def pull_recording(recording_name):
+    time.sleep(5)
+    proc = subprocess.Popen(['adb', 'pull'] + [recording_name])
+    proc.wait()
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='record video through adb', usage='record_adb.py --input <touch> --coordinate-x and --cordinate-y or <intent> --recording-name <name.mp4>')
+    parser.add_argument('-i','--input', required=True, choices=("touch", "intent"))
+    parser.add_argument('-cx','--coordinate-x', type=int, help="X position of touch event")
+    parser.add_argument('-cy','--coordinate-y', type=int, help="Y position of touch event")
+    parser.add_argument('-o','--output', type=str, help="Y position of touch event")
+    args = parser.parse_args()
+    main(args)
```

Comment:
This patch includes an mp4: I don't think we want to include that in the repo.

Reply:
should probably edit the gitignore 😛

Reply:
Can you squash the commits such that the mp4 add/remove isn't included in the history? It'll bloat the repository even if it's not in the directory at present.